lua/telescope/actions/utils.lua

```diff
@@ -0,0 +1,58 @@
+---@tag telescope.actions.utils
+
+---@brief [[
+--- Utilities to wrap functions around picker selections and entries.
+---
+--- Generally used from within other |telescope.actions|
+---@brief ]]
+
+local action_state = require('telescope.actions.state')
+
+local utils = {}
+
+--- Apply `f` to the entries of the current picker and return a table of mapped entries.
+--- - Note: mapped entries may include results not visible in the results popup.
+---@param prompt_bufnr number: The prompt bufnr
+---@param f function: Function to apply on entries of picker that takes (entry, index, row) as possible arguments
+---@return table: Result from `f` applied to entries
+function utils.map_entries(prompt_bufnr, f)
+  vim.validate{
+    f = {f, "function"}
+  }
+  local current_picker = action_state.get_current_picker(prompt_bufnr)
+  local results = {}
+  local index = 1
+  local row, result
+    -- indices are 1-indexed, rows are 0-indexed
+  for entry in current_picker.manager:iter() do
+    row = current_picker:get_row(index)
+    -- assign result to variable as function w/o return value
+    -- results in error upon table.insert as it's not technically nil but 'empty'
```

Comment:
I dont get this comment

Reply:
Like if `f` returns nil and you do `results[index] = nil` in this case. Its nil at this index. If f now only returns nil for some elements, then in the middle would be a nil -> only 1 .. middle will be iterated. Am i wrong?

Reply:
If so and a function can return nil you should do a if not nil check and put line 32 33 in that check

Reply:
See the other comment -- it takes care of functions that do not have a return statement.

Reply:
Yeah but this index will continue to insert things. Thats the point. You could get "holes" in your table, when doing it with `results[index] = result`

Reply:
```lua
local a = function()
  local results = {}

  local odd = function(i)
    if i % 2 == 1 then
      return i
    end
  end

  for i = 1, 10 do
    results[i] = odd(i)
  end

  P(results)
end

local b = function()
  local results = {}

  local odd = function(i)
    if i % 2 == 1 then
      return i
    end
  end

  for i = 1, 10 do
    local result = odd(i)
    table.insert(results, result)
  end

  P(results)
end

a()
b()
```

Reply:
> Yeah but this index will continue to insert things. Thats the point. You could get "holes" in your table, when doing it with results[index] = result

Fair enough, didn't consider it in that context, but that's also why I initially relied on table.insert ;)

Reply:
tbh right now. idk what we want :rofl: 

You can still do it with an index but you need a if not nil check. I think about all that and maybe do a coin flip ;)

Reply:
What about:
- default to just returning a table of values with `table.insert` as I had it -- that's most likely what's being used anyways
- do `index` and `result` key-value table when a tuple is returned

I don't think speed is a practical concern here, we then have a practical default, and if a user needs the index, he can have it.